src/main/java/org/jabref/gui/maintable/columns/MainTableColumnFactory.java

```diff
@@ -0,0 +1,241 @@
+     */
+    private TableColumn<BibEntryTableViewModel, ?> createGroupColumn(MainTableColumnModel columnModel) {
+        TableColumn<BibEntryTableViewModel, List<AbstractGroup>> column = new MainTableColumn<>(columnModel);
+        Node headerGraphic = IconTheme.JabRefIcons.DEFAULT_GROUP_ICON.getGraphicNode();
+        Tooltip.install(headerGraphic, new Tooltip(Localization.lang("Group color")));
+        column.setGraphic(headerGraphic);
+        column.getStyleClass().add(STYLE_ICON_COLUMN);
+        setExactWidth(column, ColumnPreferences.ICON_COLUMN_WIDTH);
+        column.setResizable(false);
+        column.setCellValueFactory(cellData -> cellData.getValue().getMatchedGroups());
+        new ValueTableCellFactory<BibEntryTableViewModel, List<AbstractGroup>>()
+                .withGraphic(this::createGroupColorRegion)
+                .install(column);
+        column.setStyle("-fx-padding: 0 0 0 0;");
+        column.setSortable(true);
+        return column;
+    }
+
+    private Node createGroupColorRegion(BibEntryTableViewModel entry, List<AbstractGroup> matchedGroups) {
+        List<Color> groupColors = matchedGroups.stream()
+                                               .flatMap(group -> OptionalUtil.toStream(group.getColor()))
+                                               .collect(Collectors.toList());
+
+        if (!groupColors.isEmpty()) {
+            HBox container = new HBox();
+            container.setSpacing(2);
+            container.setMinWidth(10);
+            container.setAlignment(Pos.CENTER_LEFT);
+            container.setPadding(new Insets(0, 2, 0, 2));
+
+            groupColors.stream().distinct().forEach(groupColor -> {
+                Rectangle groupRectangle = new Rectangle();
+                groupRectangle.getStyleClass().add("groupColumnBackground");
+                groupRectangle.setWidth(3);
+                groupRectangle.setHeight(18);
+                groupRectangle.setFill(groupColor);
+                groupRectangle.setStrokeWidth(1);
+                container.getChildren().add(groupRectangle);
+            });
+
+            String matchedGroupsString = matchedGroups.stream()
+                                                      .distinct()
+                                                      .map(AbstractGroup::getName)
+                                                      .collect(Collectors.joining(", "));
+            Tooltip tooltip = new Tooltip(Localization.lang("Entry is contained in the following groups:") + "\n" + matchedGroupsString);
+            Tooltip.install(container, tooltip);
+            return container;
+        }
+        return new Pane();
+    }
+
+    /**
+     * Creates a text column to display any standard field.
+     */
+    private TableColumn<BibEntryTableViewModel, ?> createFieldColumn(MainTableColumnModel columnModel) {
+        return new FieldColumn(columnModel, FieldFactory.parseOrFields(columnModel.getQualifier()));
```

Comment:
I think all the single-line `create` methods can now be inlined. 

For the FieldColumn, it might make sense to remove the second argument (as this is also extracted from the columnModel, and thus can be done in the constructor (or maybe move the field parsing to the columnModel and replace the `getQualifier` by `getFields`).

Reply:
I tried inlining the create* methods, but I realized that this made the constructor very hard to read. I really would prefer to keep the methods, even though they could.

The MainTableColumnModel works independently from Field, since it also displays an index column. I moved the FieldFactory.parseOrFields into the FieldColumn-class.

Reply:
Sounds good to me!